File: android-examples/01_first-app/app/src/main/java/xyz/ourguide/app/firstapp/MainActivity.java
package xyz.ourguide.app.firstapp;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

// # 기존 안드로이드 프로젝트에 코틀린 지원 추가하기.
// gradle
//    build.gradle(project)
//     - build.gradle(app)

// 1. 프로젝트 수준 build.gradle 파일에 classpath 추가
// classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.41'

// 2. 모듈(app) 수준 build.gradle 파일에 코틀린 플러그인 적용
// apply plugin: 'kotlin-android'

// 3. 모듈(app) 수준 build.gradle 파일에 의존성 추가
// implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.2.41'

// 4. java 코드와 kotlin 코드를 별도의 디렉토리로 분리해서 사용하는 것이 좋다.
/*
  sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
  }
*/


/*
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}
*/

File: android-examples/first-app/app/src/main/kotlin/xyz/ourguide/app/firstapp/MainActivity.kt
package xyz.ourguide.app.firstapp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

/*
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}
*/

// Kotlin Android Extension
//   => Extension Function
//   => 모듈 수준 build.gradle에서
//      apply plugin

/*
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val helloButton = findViewById<Button>(R.id.helloButton)
        val textView = findViewById<TextView>(R.id.textView)
        helloButton.setOnClickListener {
            textView.setText("Hello, Kotlin")
        }
    }
}
*/


// findViewById는 더 이상 사용할 필요가 없습니다.
//    => layout.xml 파일에서 설정한 view id를 바로 참조로 접근 가능합니다.

//   android:id="@+id/button_hello"  <- 이런 스타일의 이름 지정은 지양하는 것이 좋습니다.
//   android:id="@+id/helloButton"

//   Refactor - Rename
//    : Shift + F6
//    * layout 파일에서 이름을 변경하면 적용됩니다.


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        hiButton.setOnClickListener {
            textView.setText("Hello, Kotlin")
        }
    }
}























